winpr/libwinpr/utils/asn1/asn1.c

```diff
@@ -1000,7 +1000,7 @@ size_t WinPrAsn1DecReadBoolean(WinPrAsn1Decoder* dec, WinPrAsn1_BOOL* target)
 	ret = readTagAndLen(dec, &dec->source, &tag, &len);
 	if (!ret || tag != ER_TAG_BOOLEAN)
 		return 0;
-	if (!Stream_CheckAndLogRequiredLength(TAG, &dec->source, len) || len != 1)
+	if (Stream_GetRemainingLength(&dec->source) < len || len != 1)
```

Comment:
any reason for not logging a parser error here?

Reply:
When I've added negative tests (tests when input stream is invalid) I've noticed that on the MacOs CI, this emits a backtrace and it seems like the backtrace support sometime segfaults (not always). And also I'm not sure that super verbose log really helps, as the higher level message saying "can't read message XXXX (containing a bool or an int)" is usually enough. That also makes the unitary test more quiet.

Reply:
so you try to hide some bug on mac os?
as for logging, it is usually quite helpful, specifically if you know where the issue occurred.

Reply:
Well logging at the lowest level here never convinced me, it's really too much to verbosely log and print stack traces for "just a malformed piece of ASN1 message". Initial implementation of ASN1 functions were not that verbose. These kind of logs do help when parsing RDP packets, not with lower level items as ASN1.
BTW we lack some tests on invalid content on these, the fuzzing tests kinda help as they do it for us, but perhaps I should add more negative tests for ASN1 parsing.

But yes you're right, the one main reason was to make the Mac CI happy as I couldn't reproduce here, and it doesn't always happen.